### PR TITLE
[WIP] Add tab-completion for powershell

### PR DESCRIPTION
**Pull request by kblohm**
Hi,
I tried to add some basic tab-completion for powershell.
This does probably not cover all the possibilities, but as far as i can tell it works fine for everything i tried.
Sometimes i feel like the fake-cli help is not super useful, so i might have missed some things.
One other painpoint is completion for targets. The --list argument needs to first download and build everything, so it might take a little while when he build never ran before.

I am not very firm in powershell-scripting, so i dont know if this is the right approach. I just looked how other people did something similair. Feedback is appreciated :).
There is no documentation yet, because i am not sure how useful this is or if you even want this in this repo. I could add some later though.

### CODE REVIEW

**Comment by matthid** [comment]
I try to not use powershell as much as possible, but I'm fine with adding this to this repository. 

But we definitely would need a page to explain how people can use this. Also can we somehow ensure that if we add new CLI arguments we do not forget to update this powershell script?

**Comment by kblohm** [comment]
I dont think there is an easy way to enforce that. The approach that the dotnet-cli is taking is kinda neat. The library that parses the commandline-arguments also creates suggestions, so the shell scripts only have to ask for them. (https://github.com/dotnet/CliCommandLineParser)
That would require quite a lot of changes though. Especially calling the Target-Api from the script-runner.
It has the benefit of working for bash aswell.

So in short, i guess that is just discipline, like adding documentation for modules.
As far as i can tell, the bash-completion (if that is even used) is also not very up to date.

**Comment by BlythMeister** [comment]
Hmm I guess I'm more concerned about repeatable builds because of legacy branching rather than trunk based.

**Comment by matthid** [comment]
@BlythMeister I guess I differentiate between CI and local builds. Locally it is nice to just `fake build` in the shell, on CI it will use always the same fake-runtime version. But I don't really want to steal this issue, which is about powershell completion. We can discuss this in another issue if needed.

**Comment by kblohm** [comment]
Sorry, i forgot to write a comment after i pushed.
I wanted to add a bash-version. But i can also make another PR for that if you want to merge.

**Comment by kblohm** [comment]
I added a completion file for bash. I could only test it in git-bash though.  It would be nice if someone could have another look. I am also not 100% sure if the installation instructions work on every system, because i could not test them :/.

**Comment by kblohm** [comment]
I had a look at the git completion where you also have to write -- to complete arguments. Otherwise it will default to commands e.g add, commit etc. I am not sure what the UNIX default is there. When i write fake build target and double tab i get a list of targets. That list is somewhat wrong though :/. I forgot to handle targets with descriptions.

**Comment by matthid** [comment]
@kblohm Yes sorry just tested it again and it works, I just was in another directory and then you just get the default completion (list of files)
